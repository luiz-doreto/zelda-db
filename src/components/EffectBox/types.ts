export type EffectBoxProps = {
  value: string | number;
  type: 'attack' | 'defense' | 'bonus';
};
