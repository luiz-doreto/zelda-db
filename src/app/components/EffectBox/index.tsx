import { StyleSheet, Text, View } from 'react-native';
import Shield from '~/assets/images/shield.svg';
import Sword from '~/assets/images/sword.svg';
import { EffectBoxProps } from './types';

const EffectBox = ({ value, type }: EffectBoxProps) => {
  const renderEffectIcon = () =>
    ({
      attack: <Sword height={16} width={16} />,
      defense: <Shield height={16} width={16} />,
      bonus: null,
    })[type];

  return (
    <View style={styles.container}>
      {renderEffectIcon()}
      <View
        style={[
          styles.propContainer,
          type !== 'bonus' ? styles.iconMargin : null,
        ]}
      >
        <View style={styles.innerPropContainer}>
          <Text style={styles.value}>{value}</Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    marginRight: 8,
  },
  iconMargin: {
    marginLeft: 6,
  },
  propContainer: {
    minWidth: 35,
    width: 'auto',
    height: 25,
    borderRadius: 3,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 2,
  },
  innerPropContainer: {
    width: '100%',
    height: '100%',
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: 'rgba(299, 299, 299, 0.2)',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 3,
  },
  value: {
    color: 'white',
  },
});

export default EffectBox;
