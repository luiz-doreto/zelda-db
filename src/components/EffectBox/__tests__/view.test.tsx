import { render } from '@testing-library/react-native';
import EffectBox from '..';

describe('EffectBox', () => {
  it('should render correctly', () => {
    const { getByText } = render(<EffectBox value={10} type="attack" />);
    expect(getByText('10')).toBeTruthy();
  });

  it('should render correctly with defense type', () => {
    const { getByText } = render(<EffectBox value={10} type="defense" />);
    expect(getByText('10')).toBeTruthy();
  });

  it('should render correctly with bonus type', () => {
    const { getByText } = render(<EffectBox value={10} type="bonus" />);
    expect(getByText('10')).toBeTruthy();
  });

  it('should match snapshot for attack type', () => {
    const { toJSON } = render(<EffectBox value={10} type="attack" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for defense type', () => {
    const { toJSON } = render(<EffectBox value={10} type="defense" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot for bonus type', () => {
    const { toJSON } = render(<EffectBox value={10} type="bonus" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
