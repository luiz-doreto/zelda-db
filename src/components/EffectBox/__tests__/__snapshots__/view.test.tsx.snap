// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EffectBox should match snapshot for attack type 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginRight": 8,
    }
  }
>
  <SvgMock
    height={16}
    width={16}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 3,
          "height": 25,
          "justifyContent": "center",
          "minWidth": 35,
          "padding": 2,
          "width": "auto",
        },
        {
          "marginLeft": 6,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "rgba(299, 299, 299, 0.2)",
          "borderWidth": 1,
          "height": "100%",
          "justifyContent": "center",
          "paddingHorizontal": 3,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      >
        10
      </Text>
    </View>
  </View>
</View>
`;

exports[`EffectBox should match snapshot for bonus type 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginRight": 8,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 3,
          "height": 25,
          "justifyContent": "center",
          "minWidth": 35,
          "padding": 2,
          "width": "auto",
        },
        null,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "rgba(299, 299, 299, 0.2)",
          "borderWidth": 1,
          "height": "100%",
          "justifyContent": "center",
          "paddingHorizontal": 3,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      >
        10
      </Text>
    </View>
  </View>
</View>
`;

exports[`EffectBox should match snapshot for defense type 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginRight": 8,
    }
  }
>
  <SvgMock
    height={16}
    width={16}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 3,
          "height": 25,
          "justifyContent": "center",
          "minWidth": 35,
          "padding": 2,
          "width": "auto",
        },
        {
          "marginLeft": 6,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "rgba(299, 299, 299, 0.2)",
          "borderWidth": 1,
          "height": "100%",
          "justifyContent": "center",
          "paddingHorizontal": 3,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
          }
        }
      >
        10
      </Text>
    </View>
  </View>
</View>
`;
